src/components/shared/Map/MapPointer.tsx

```diff
@@ -4,35 +4,51 @@ import { Circle, Marker, Popup } from 'react-leaflet';
 
 import { type SensorData } from '@/types/models/SensorData';
 
+import { COLORS, LEVELS } from '../../../constants/index';
 import PointerPopup from './PointerPopup';
 
+type PollutionType = 'PM2.5' | 'PM10';
+
 interface MapPointerProps {
   center: [number, number];
   radius: number;
   text: string;
-  color: string;
   sensorData: SensorData;
 }
 
-function MapPointer({
-  center,
-  radius,
-  text,
-  color,
-  sensorData
-}: MapPointerProps) {
+function getCircleColor(
+  sensorData: SensorData,
+  pollutionName: PollutionType
+): string {
+  if (pollutionName != null) {
+    const pollutionValue = sensorData.data.find(
+      item => item.name === pollutionName
+    );
```

Comment:
I don't like the current `SensorData` type implementation as it requires checking if the measurement exists. If we are sure that there always will be one PM2.5 and one PM10 measurement, the `SensorData` might look like this:
```ts
export interface SensorData {
  sensorName: string;
  measurementDate: Date;
  measurements: {
    'PM2.5': number;
    'PM10': number;
  };
}
```
I know that you haven't implemented this interface but it is a bit misleading suggesting that there will be a variable number of measurements.

After that you can change the `getCircleColor` to the following implementation:
```ts
const getCircleColor = (
  sensorData: SensorData,
  pollutionName: PollutionType
): string => {
  const pollutionValue = sensorData.measurements[pollutionName];
  const thresholds = POLLUTION.pollutionLevels[pollutionName];
  const idx = thresholds.findIndex(
    (threshold: number) => pollutionValue < threshold
  );
  return COLORS.pollutionScale[idx] as string;
};
```
I removed `as any` type casts. We shouldn't use `any` type as it makes TypeScript useless. You also use imported constants incorrectly and, because of casting type to `any`, you don't get a hint that something is wrong. For example, `COLORS` is the following object:
```ts
{
  pollutionScale: [
    '#6ebc2a',
    '#c4d83f',
    '#ffe500',
    '#f1aa41',
    '#b91414',
    '#590663',
    '#909090'
  ]
}
```
but you expect it to be an array which will not work.

Reply:
Firstly, thanks for noticing that one part was not my code. Secondly, great, I get the idea although it was marked in red and expecting me to cast it. I will of course consider everything.

Reply:
To be honest the SensorData part should be talked out with someone else, who can say more about this structure and usages because I am only in charge of colors. I does not change much in my case and I am just working with provided sensorData structure. It should be more appropriate for people who need to work more with its contents. I can change it according to your suggestions, but we should be sure that these changes are reasonable. Or, if this is actually only important for my part, I could just change it, but I want to be sure.

Reply:
In addition, about the bad imports, it is my first encounter with typescript ever, the Internet was not very helpful and this was my only way around.